File: debug_products.js
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  // Start a local server in the background
  const { exec } = require('child_process');
  const server = exec('python3 -m http.server 8081');
  
  await new Promise(resolve => setTimeout(resolve, 2000)); // Wait for server to start
  
  try {
    await page.goto('http://localhost:8081/index.html');
    await page.waitForTimeout(2000); // Wait for fetch
    
    const productCardsCount = await page.locator('.bg-white.rounded-2xl').count();
    console.log(`Found ${productCardsCount} product cards.`);
    
    const html = await page.content();
    if (html.includes('Loading...')) {
        console.log("Still loading...");
    }
    
    await page.screenshot({ path: '/home/jules/verification/debug_products.png', fullPage: true });
    
    // Check for errors in console
    page.on('console', msg => console.log('PAGE LOG:', msg.text()));

  } catch (error) {
    console.error('Error:', error);
  } finally {
    await browser.close();
    server.kill();
  }
})();

File: debug_products.py
import asyncio
from playwright.async_api import async_playwright
import subprocess
import time
import os

async def main():
    # Start server
    server = subprocess.Popen(['python3', '-m', 'http.server', '8081'])
    time.sleep(2)
    
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        
        try:
            await page.goto('http://localhost:8081/index.html')
            # Wait for data to load
            await page.wait_for_timeout(3000)
            
            cards = await page.query_selector_all('.bg-white.rounded-2xl')
            print(f"Found {len(cards)} cards")
            
            for i, card in enumerate(cards[:5]):
                text = await card.inner_text()
                print(f"Card {i} text: {text[:50]}...")

            await page.screenshot(path='/home/jules/verification/debug_products_py.png', full_page=True)
            
            # Check if menu.json was loaded
            # Check network requests
            
        except Exception as e:
            print(f"Error: {e}")
        finally:
            await browser.close()
            server.terminate()

if __name__ == '__main__':
    asyncio.run(main())
